common/djangoapps/course_modes/views.py

```diff
@@ -202,13 +203,18 @@ def get(self, request, course_id, error=None):
                 for x in verified_mode.suggested_prices.split(",")
                 if x.strip()
             ]
+            price_after_discount = None
             price_before_discount = verified_mode.min_price
+            if is_enterprise_learner(request.user) and verified_mode.sku:
```

Comment:
I think we should use more optimal method `enterprise_customer_for_request` to determine if user is linked to an enterprise or not. Something like this.
```
enterprise_customer = enterprise_api.enterprise_customer_for_request(request)
if enterprise_customer and verified_mode.sku:
```